### PR TITLE
pkg/kubelet: clean up pod manager

### PR DESCRIPTION
**Pull request by xiang90**
A bunch of clean up

### CODE REVIEW

File: pkg/kubelet/pod_manager.go [hunk 1]
```diff
@@ -127,13 +127,13 @@ func (self *basicPodManager) UpdatePods(u PodUpdate, podSyncTypes map[types.UID]
 }
 
 // Set the internal pods based on the new pods.
-func (self *basicPodManager) SetPods(newPods []api.Pod) {
-	self.lock.Lock()
-	defer self.lock.Unlock()
-	self.setPods(newPods)
+func (pm *basicPodManager) SetPods(newPods []api.Pod) {
+	pm.lock.Lock()
```
Comment (yifan-gu) [inline pkg/kubelet/pod_manager.go]: Unrelated to this PR, but how about pm.Lock()?

**Comment by vmarmol** [comment]
Why the change from self BTW?

**Comment by xiang90** [comment]
@vmarmol 

see https://github.com/golang/go/wiki/CodeReviewComments#receiver-names 

I hope we can keep it go-style in kubelet pkg at least.

File: pkg/kubelet/pod_manager.go (hunk 1 above)
Comment (xiang90) [inline pkg/kubelet/pod_manager.go]: We need to be careful about using public embedded field. If we embed lock, lock would be a public field. People can call pm.Lock() outside the pkg.

**Comment by vmarmol** [comment]
Sad, one of the places I don't agree with the Go style guide...I am not one to go against the style guide though :)

**Comment by xiang90** [comment]
@vmarmol Hm... 

I usually think in this way: in the middle of the code block, the reader knows better what`pm`(pod manager) refers to than `self`.

Also when you want to convert a method on a receiver to a function with the previous receiver as arg, it can reduce your work. 

But anyway...

**Comment by vmarmol** [comment]
For the lock, DeletePod() is a remote request to the apiserver. We should probably not block on that.

**Comment by xiang90** [comment]
@vmarmol OK. So RLock is not good enough. I will address the locking issue.

**Comment by vmarmol** [comment]
@xiang90, I kinda think the opposite actually :) `pm` is a different variable on each struct whereas `self` is always the same. I know what `self` is no matter where I place it, whereas if I am dumped into a random method, I need to lookup what `pm` means. Methods shouldn't be long enough that the declaration is not clearly visible, but that's another matter.

That's true on the conversion point. I haven't personally ran into that too often, but others may have.

The style guide is the style guide though and I'd rather grumpily write `pm` and be consistent accross the codebase :D

**Comment by yujuhong** [comment]
@xiang90, thanks for the cleanup. LGTM overall, pending on rebasing to fix the lock issue.

I agree with @vmarmol that `self` is easier for me to know the context when navigating the codebase. On the other hand, I wish it was shorter :) I can be persuaded either way, but if we decide to follow the style guide, I'd prefer having one PR changing all the files (or at least files under kubelet).

**Comment by xiang90** [comment]
@yujuhong Hrm.. I can revert the first commit. I will do it later today. Thanks for the review!

**Comment by xiang90** [comment]
@vmarmol @yujuhong Rebased. I will address the `self` thing in another pr.

**Comment by yujuhong** [comment]
LGTM. Will merge on green (@vmarmol is OOO).

**Pull request merged**